kuama_permissions/lib/src/blocs/_permissions_state.dart

```diff
@@ -1,59 +1,277 @@
 part of 'permissions_bloc.b.dart';
 
-enum PermissionPlace { checking, asking, requesting }
+@DataClass(copyable: true)
+abstract class PermissionsBlocState with _$PermissionsBlocState {
+  /// The status is updating. Any iteration with the bloc is blocked.
+  ///
+  /// This normally happens when the app is brought back to the foreground.
+  /// It can lead to the resolution of a request confirmation when all the permissions of it are guaranteed.
```

Comment:
```suggestion
  /// Whether the status is updating, any iterations with the bloc are blocked when true.
  ///
  /// This normally happens when the app has been brought back to the foreground.
  /// It can lead to the resolution of a request confirmation when all its permissions are guaranteed.
```